clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -395,6 +403,8 @@ public synchronized void maybeAddPartition(TopicPartition topicPartition) {
                     " to transaction while in state  " + currentState);
             } else if (isPartitionAdded(topicPartition) || isPartitionPendingAdd(topicPartition)) {
                 return;
+            } else if (coordinatorSupportsTransactionV2) {
+                txnPartitionMap.getOrCreate(topicPartition);
```

Comment:
I think this should be fine, what would be the problem?  With transactional protocol v2 we don't really need to keep track on the client of what partitions were added, so maybe we shouldn't even add it to `partitionsInTransaction` set, just make sure that we don't block sends for partitions that are not present there.   Also, just because we've got a failure from a produce request doesn't mean we didn't add a partition, on failure we don't know for sure.

Reply:
That makes sense, I can address this in the next update.

Reply:
This is addressed by line 422?

Reply:
Yes, partitions are always added if the V2 is enabled.